### PR TITLE
[Catalog] Remove OAuth proxy

### PR DESCRIPTION
**Pull request by adutra**

### CODE REVIEW

**Review by snazy — COMMENTED** [review]
Just a comment about providing users some guidance with the error message.

File: catalog/service/rest/src/main/java/org/projectnessie/catalog/service/rest/IcebergApiV1GenericResource.java [hunk 1]
```diff
@@ -123,6 +124,19 @@ public IcebergS3SignResponse s3sign(
...
 
+  @POST
+  @Path("/v1/oauth/tokens")
+  @Consumes(MediaType.APPLICATION_FORM_URLENCODED)
+  @PermitAll
+  public Response getToken() {
+    return Response.status(Response.Status.SERVICE_UNAVAILABLE)
+        .entity(
+            Map.of(
+                "error", "OAuthTokenEndpointUnavailable",
```
Comment (dimas-b) [inline catalog/service/rest/src/main/java/org/projectnessie/catalog/service/rest/IcebergApiV1GenericResource.java:135]: Why not give a response in `IcebergApiV1AuthResource`?

File: catalog/service/rest/src/main/java/org/projectnessie/catalog/service/rest/IcebergApiV1GenericResource.java [hunk 2]
```diff
@@ -123,6 +124,19 @@ public IcebergS3SignResponse s3sign(
     return icebergS3SignResponse(signed.uri().toString(), signed.headers());
   }
 
+  @POST
+  @Path("/v1/oauth/tokens")
+  @Consumes(MediaType.APPLICATION_FORM_URLENCODED)
+  @PermitAll
+  public Response getToken() {
+    return Response.status(Response.Status.SERVICE_UNAVAILABLE)
```
Comment (dimas-b) [inline catalog/service/rest/src/main/java/org/projectnessie/catalog/service/rest/IcebergApiV1GenericResource.java:132]: 503 is supposed to be transient... how about 501 (Not Implemented)?

File: catalog/service/rest/src/main/java/org/projectnessie/catalog/service/rest/IcebergApiV1GenericResource.java (hunk 1 above)
Comment (adutra) [reply catalog/service/rest/src/main/java/org/projectnessie/catalog/service/rest/IcebergApiV1GenericResource.java]: That file is deleted. The endpoint is now so dumb that I figured it didn't make sense to put it in its own file.

**Review by snazy — DISMISSED** [review]

File: catalog/service/rest/src/main/java/org/projectnessie/catalog/service/rest/IcebergApiV1GenericResource.java (hunk 1 above)
Comment (dimas-b) [reply catalog/service/rest/src/main/java/org/projectnessie/catalog/service/rest/IcebergApiV1GenericResource.java]: but will this method be called for other (sub-) paths?

Comment (dimas-b) [reply catalog/service/rest/src/main/java/org/projectnessie/catalog/service/rest/IcebergApiV1GenericResource.java]: Should we allow this to match `:ref/v1/oauth/tokens`?

**Review by dimas-b — COMMENTED** [review]
LGTM, but I wonder whether we can/should send 501 for `":ref/v1/oauth/tokens` too.

**Comment by adutra** [comment]
> LGTM, but I wonder whether we can/should send 501 for `":ref/v1/oauth/tokens` too.

OK I think you have a point, let's add that endpoint.

**Review by dimas-b — APPROVED** [review]

**Pull request merged**